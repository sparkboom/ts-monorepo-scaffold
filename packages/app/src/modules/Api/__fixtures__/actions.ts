export const persistRehydrateWithToken = {
  type: 'persist/REHYDRATE',
  payload: {
    api: {
      token: {
        jwt: 'PERSISTED_TOKEN',
        expiryTimeStamp: 1571615706078,
        expiresInSeconds: 3311.836,
      },
    },
    _persist: {
      version: -1,
      rehydrated: true,
    },
  },
  key: '%%APP_NAME%%',
};

export const persistRehydrateNoToken = {
  type: 'persist/REHYDRATE',
  payload: {
    api: {
      token: {},
    },
    _persist: {
      version: -1,
      rehydrated: true,
    },
  },
  key: '%%APP_NAME%%',
};
