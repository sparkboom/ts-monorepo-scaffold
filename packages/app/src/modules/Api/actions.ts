import { Action } from 'redux';

// Types
interface ApiSetTokenAction {
  (jwt: string, expiryTimeStamp: number): AppAction<'API__SET_TOKEN', { jwt: string; expiryTimeStamp: number }>;
}
interface ApiInitAction {
  (): Action<'API__INIT'>;
}

// Actions
export const apiSetToken: ApiSetTokenAction = (jwt, expiryTimeStamp) => ({
  type: 'API__SET_TOKEN',
  payload: {
    jwt,
    expiryTimeStamp,
  },
});

export const apiInit: ApiInitAction = () => ({
  type: 'API__INIT',
});
