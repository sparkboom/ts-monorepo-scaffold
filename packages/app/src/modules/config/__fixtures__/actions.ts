export const fullConfiguration = {
  tasks: {
    taskUpdateFrequencyMs: 10000,
  },
  featureToggles: {},
  endpoints: {
    query: {
      baseUrl: 'http://query.example.com/',
    },
    command: {
      baseUrl: 'http://cmd.example.com/',
    },
  },
};

export const partialConfiguration = {
  tasks: {
    taskUpdateFrequencyMs: 10000,
  },
  featureToggles: {},
  endpoints: {
    query: {
      baseUrl: 'http://query.example.com/',
    },
    command: {
      baseUrl: 'http://cmd.example.com/',
    },
  },
};

export const fullFeatureToggles = {
  supportMultiSelect: true,
  supportColumnSort: true,
  supportBulkReAuth: true,
  supportBulkDelete: true,
  supportFilter: true,
  supportSearch: true,
};

export const partialFeatureToggles = {
  supportMultiSelect: false,
  supportColumnSort: false,
  supportSearch: false,
};
