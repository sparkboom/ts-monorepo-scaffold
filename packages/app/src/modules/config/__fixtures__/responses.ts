export const salescloudCapabilities = {
  leadIngestionTask: {
    destination: {
      marketingCloud: {
        dataExtension: false,
      },
      salesCloud: {
        leadObject: true,
      },
    },
  },
};

export const marketingcloudCapabilities = {
  leadIngestionTask: {
    destination: {
      marketingCloud: {
        dataExtension: true,
      },
      salesCloud: {
        leadObject: false,
      },
    },
  },
};
