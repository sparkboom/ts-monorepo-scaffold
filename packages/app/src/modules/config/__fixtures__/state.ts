import { LeadCapConfig } from '../types';

// State
export const simpleState: Partial<LeadCapConfig> = {
  tasks: {
    taskUpdateFrequencyMs: 10,
  },
  featureToggles: {
    supportMultiSelect: true,
    supportColumnSort: true,
    supportReAuthAllTasks: true,
    supportBulkDelete: true,
    supportFilter: true,
    supportSearch: true,
  },
  endpoints: {
    command: {
      baseUrl: 'http://cmd-leadcap.example.com',
    },
    query: {
      baseUrl: 'http://query-leadcap.example.com',
    },
    web: {
      baseUrl: 'http://web-leadcap.example.com',
      basePath: '',
    },
  },
  platform: null,
};

export const emptyState: Partial<LeadCapConfig> = {};
