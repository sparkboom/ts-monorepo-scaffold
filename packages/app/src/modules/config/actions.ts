/* eslint-disable import/prefer-default-export */
import { Action, DeepPartial } from 'redux';
import { LeadCapConfig, FeatureToggles } from './types';

// Types
type SetContent = (config: DeepPartial<LeadCapConfig>) => AppAction<'CONFIG__SET', DeepPartial<LeadCapConfig>>;
type SetTasks = (toggles: FeatureToggles) => AppAction<'CONFIG__FEATURE_TOGGLES_SET', { featureToggles: FeatureToggles }>;

// Actions
export const setConfig: SetContent = config => ({
  type: 'CONFIG__SET',
  payload: config,
});

export const loadConfig = (): Action<'CONFIG__LOAD'> => ({
  type: 'CONFIG__LOAD',
});

export const setFeatureToggles: SetTasks = toggles => ({
  type: 'CONFIG__FEATURE_TOGGLES_SET',
  payload: {
    featureToggles: toggles,
  },
});

export const loadFeatureToggles = (): Action<'CONFIG__FEATURE_TOGGLES_LOAD'> => ({
  type: 'CONFIG__FEATURE_TOGGLES_LOAD',
});
