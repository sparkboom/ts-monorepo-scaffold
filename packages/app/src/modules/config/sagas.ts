import { takeLatest, all, apply, call, select } from 'redux-saga/effects';
import { ClientWebApi, ClientQueryApi, ApiCapabilities } from '@adstudio-leadcap/api';
import { Action } from 'redux';
import { get } from 'lodash-es';
import { setConfig, setFeatureToggles } from './actions';
import { correlate } from '../../middleware/correlate-redux-middleware/index';
import { status, success, fail, submitError } from '../Logging/actions';
import { Platform } from './types';
import { getWebApi, getQueryApi } from '../Api/sagas';
import { putContext } from '../Logging/effects';

// Constants
const LOG_CTX = 'CONFIG.SAGAS';

// Sagas
export function* loadConfigSaga(action: Action<any>, global: any = window) {
  const put = putContext(action, LOG_CTX);

  // 1. Set config set-up to 'in progress'
  yield put(status(action, 'IN_PROGRESS'));

  // 2. Get Web API client
  const webApi = yield call(getWebApi, action);
  if (!webApi) {
    // TODO: display toast to notify
    yield put(fail(action));
    return;
  }

  // 3. Load configuration file
  const config = yield apply(webApi, webApi.config, []);
  if (config) {
    yield put(setConfig(config));
  } else {
    yield put.errorWarn('Configuration could not be downloaded');
    // TODO: provide error reason status
    yield put(fail(action));
    return;
  }

  // 4. Get Query API client
  const queryApi = yield call(getQueryApi, action);
  if (!queryApi) {
    // TODO: display toast to notify
    yield put(fail(action));
    return;
  }

  // 5. Store Platform
  if (global === global.top) {
    yield put(setConfig({ platform: 'STAND_ALONE' }));
  } else {
    // TODO: Capabilities endpoint is perhaps overkill unless they are permissions checks, can we pass 'platform' property from log-in endpoint?
    // eslint-disable-next-line @typescript-eslint/unbound-method
    const capabilities: ApiCapabilities = yield apply(queryApi, queryApi.getCapabilities, []);
    let platform = 'UNKNOWN';
    if (get(capabilities, 'leadIngestionTask.destination.marketingCloud.dataExtension') === true) {
      platform = 'MARKETING_CLOUD';
    } else if (get(capabilities, 'leadIngestionTask.destination.salesCloud.leadObject') === true) {
      platform = 'SALES_CLOUD';
    }

    yield put(setConfig({ platform }));

    if (platform === 'UNKNOWN') {
      yield put.errorWarn('Platform could not be determined (i.e. neither MARKETING_CLOUD nor SALES_CLOUD )');
      yield put(fail(action));
      return;
    }
  }

  // 7. Mark config load a successful
  yield put(success(action));
}

export function* loadFeatureTogglesSaga(action: Action<any>) {
  const put = putContext(action, LOG_CTX);

  const webApi = yield call(getWebApi, action);
  if (!webApi) {
    return;
  }
  const featureToggles = yield apply(webApi, webApi.featureToggles, []);
  if (featureToggles) {
    yield put(setFeatureToggles(featureToggles));
  }
}

function* configSaga() {
  yield all([takeLatest('CONFIG__LOAD', loadConfigSaga), takeLatest('CONFIG__FEATURE_TOGGLES_LOAD', loadFeatureTogglesSaga)]);
}

export default configSaga;
