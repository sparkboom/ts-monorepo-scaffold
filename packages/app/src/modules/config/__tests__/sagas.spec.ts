import { expectSaga } from 'redux-saga-test-plan';
import * as matchers from 'redux-saga-test-plan/matchers';
import * as sagas from '../sagas';
import * as actions from '../actions';
import * as responses from '../__fixtures__/responses';
import { getWebApi, getQueryApi } from '../../Api/sagas';
import { status, fail, success } from '../../Logging/actions';
import { correlate } from '../../../middleware/correlate-redux-middleware';

describe('Config > Sagas', () => {
  describe('loadConfigSaga', () => {
    let FAKE_ACTION: any;
    let FAKE_GLOBAL: any;
    let FAKE_CONFIG: any;

    beforeEach(() => {
      FAKE_ACTION = correlate(actions.loadConfig(), 'CORRELATION_ID');
      FAKE_GLOBAL = {};
      FAKE_CONFIG = {
        option1: true,
      };
    });

    it('should set configuration check to `in progress`', () => {
      // Arrange
      // Act
      // Assert
      return expectSaga(sagas.loadConfigSaga, FAKE_ACTION, FAKE_GLOBAL)
        .put(status(FAKE_ACTION, 'IN_PROGRESS'))
        .run();
    });

    it('should fail if the Web API client could not be retrieved', () => {
      // Arrange
      const FAKE_NULL_WEB_API = null;

      // Act
      // Assert
      return expectSaga(sagas.loadConfigSaga, FAKE_ACTION, FAKE_GLOBAL)
        .provide([[matchers.call.fn(getWebApi), FAKE_NULL_WEB_API]])
        .put(fail(FAKE_ACTION))
        .run();
    });

    it('should load the configuration file if the Web API client is available', () => {
      // Arrange
      const FAKE_WEB_API = {
        config: () => FAKE_CONFIG,
      };

      // Act
      // Assert
      return expectSaga(sagas.loadConfigSaga, FAKE_ACTION, FAKE_GLOBAL)
        .provide([[matchers.call.fn(getWebApi), FAKE_WEB_API], [matchers.call.fn(FAKE_WEB_API.config), FAKE_CONFIG], [matchers.call.fn(getQueryApi), null]])
        .call.fn(FAKE_WEB_API.config)
        .run();
    });

    it('should fail if the Query API client could not be retrieved', () => {
      // Arrange
      const FAKE_WEB_API = {
        config: () => FAKE_CONFIG,
      };

      // Act
      // Assert
      return expectSaga(sagas.loadConfigSaga, FAKE_ACTION, FAKE_GLOBAL)
        .provide([[matchers.call.fn(getWebApi), FAKE_WEB_API], [matchers.call.fn(FAKE_WEB_API.config), FAKE_CONFIG], [matchers.call.fn(getQueryApi), null]])
        .put(fail(FAKE_ACTION))
        .run();
    });

    it('should call getCapabilites if Query API client is available', () => {
      // Arrange
      const FAKE_WEB_API = {
        config: () => FAKE_CONFIG,
      };
      const FAKE_QUERY_API = {
        getCapabilities: () => responses.marketingcloudCapabilities,
      };

      // Act
      // Assert
      return expectSaga(sagas.loadConfigSaga, FAKE_ACTION, FAKE_GLOBAL)
        .provide([
          [matchers.call.fn(getWebApi), FAKE_WEB_API],
          [matchers.call.fn(FAKE_WEB_API.config), FAKE_CONFIG],
          [matchers.call.fn(getQueryApi), FAKE_QUERY_API],
          [matchers.call.fn(FAKE_QUERY_API.getCapabilities), responses.marketingcloudCapabilities],
        ])
        .call.fn(FAKE_QUERY_API.getCapabilities)
        .run();
    });

    it('should set the platform to `STAND_ALONE` if there is no container global', () => {
      // Arrange
      const FAKE_WEB_API = {
        config: () => FAKE_CONFIG,
      };
      const FAKE_QUERY_API = {
        getCapabilities: () => responses.marketingcloudCapabilities,
      };
      const FAKE_GLOBAL_STAND_ALONE: any = {};
      FAKE_GLOBAL_STAND_ALONE.top = FAKE_GLOBAL_STAND_ALONE;

      // Act
      // Assert
      return expectSaga(sagas.loadConfigSaga, FAKE_ACTION, FAKE_GLOBAL_STAND_ALONE)
        .provide([
          [matchers.call.fn(getWebApi), FAKE_WEB_API],
          [matchers.call.fn(FAKE_WEB_API.config), FAKE_CONFIG],
          [matchers.call.fn(getQueryApi), FAKE_QUERY_API],
          [matchers.call.fn(FAKE_QUERY_API.getCapabilities), responses.marketingcloudCapabilities],
        ])
        .put.like({ action: actions.setConfig({ platform: 'STAND_ALONE' }) })
        .run();
    });

    it('should set the platform to `MARKETING_CLOUD` if the capabilities endpoint returns true for dataExtensions', () => {
      // Arrange
      const FAKE_WEB_API = {
        config: () => FAKE_CONFIG,
      };
      const FAKE_QUERY_API = {
        getCapabilities: () => responses.marketingcloudCapabilities,
      };

      // Act
      // Assert
      return expectSaga(sagas.loadConfigSaga, FAKE_ACTION, FAKE_GLOBAL)
        .provide([
          [matchers.call.fn(getWebApi), FAKE_WEB_API],
          [matchers.call.fn(FAKE_WEB_API.config), FAKE_CONFIG],
          [matchers.call.fn(getQueryApi), FAKE_QUERY_API],
          [matchers.call.fn(FAKE_QUERY_API.getCapabilities), responses.marketingcloudCapabilities],
        ])
        .put.like({ action: actions.setConfig({ platform: 'MARKETING_CLOUD' }) })
        .run();
    });

    it('should set the platform to `SALES_CLOUD` if the capabilities endpoint returns true for leadObject', () => {
      // Arrange
      const FAKE_WEB_API = {
        config: () => FAKE_CONFIG,
      };
      const FAKE_QUERY_API = {
        getCapabilities: () => responses.salescloudCapabilities,
      };

      // Act
      // Assert
      return expectSaga(sagas.loadConfigSaga, FAKE_ACTION, FAKE_GLOBAL)
        .provide([
          [matchers.call.fn(getWebApi), FAKE_WEB_API],
          [matchers.call.fn(FAKE_WEB_API.config), FAKE_CONFIG],
          [matchers.call.fn(getQueryApi), FAKE_QUERY_API],
          [matchers.call.fn(FAKE_QUERY_API.getCapabilities), responses.salescloudCapabilities],
        ])
        .put.like({ action: actions.setConfig({ platform: 'SALES_CLOUD' }) })
        .run();
    });

    it('should set the platform to `UNKNOWN` if the capabilities endpoint returns fasle for both dataExtensions and leadObject', () => {
      // Arrange
      const FAKE_WEB_API = {
        config: () => FAKE_CONFIG,
      };
      const FAKE_QUERY_API = {
        getCapabilities: () => {},
      };

      // Act
      // Assert
      return expectSaga(sagas.loadConfigSaga, FAKE_ACTION, FAKE_GLOBAL)
        .provide([
          [matchers.call.fn(getWebApi), FAKE_WEB_API],
          [matchers.call.fn(FAKE_WEB_API.config), FAKE_CONFIG],
          [matchers.call.fn(getQueryApi), FAKE_QUERY_API],
          [matchers.call.fn(FAKE_QUERY_API.getCapabilities), {}],
        ])
        .put.like({ action: actions.setConfig({ platform: 'UNKNOWN' }) })
        .run();
    });

    it('should dispatch a success action if configuration was set', () => {
      // Arrange
      const FAKE_WEB_API = {
        config: () => FAKE_CONFIG,
      };
      const FAKE_QUERY_API = {
        getCapabilities: () => responses.marketingcloudCapabilities,
      };

      // Act
      // Assert
      return expectSaga(sagas.loadConfigSaga, FAKE_ACTION, FAKE_GLOBAL)
        .provide([
          [matchers.call.fn(getWebApi), FAKE_WEB_API],
          [matchers.call.fn(FAKE_WEB_API.config), FAKE_CONFIG],
          [matchers.call.fn(getQueryApi), FAKE_QUERY_API],
          [matchers.call.fn(FAKE_QUERY_API.getCapabilities), responses.marketingcloudCapabilities],
        ])
        .put.like({ action: success(FAKE_ACTION) })
        .run();
    });

    it('should match snapshot for Marketing Cloud platform', async () => {
      // Arrange
      const FAKE_WEB_API = {
        config: () => FAKE_CONFIG,
      };
      const FAKE_QUERY_API = {
        getCapabilities: () => responses.marketingcloudCapabilities,
      };

      // Act
      // Assert
      const result = await expectSaga(sagas.loadConfigSaga, FAKE_ACTION, FAKE_GLOBAL)
        .provide([
          [matchers.call.fn(getWebApi), FAKE_WEB_API],
          [matchers.call.fn(FAKE_WEB_API.config), FAKE_CONFIG],
          [matchers.call.fn(getQueryApi), FAKE_QUERY_API],
          [matchers.call.fn(FAKE_QUERY_API.getCapabilities), responses.marketingcloudCapabilities],
        ])
        .run();

      expect(result.toJSON()).toMatchSnapshot();
    });

    it('should match snapshot for Sales Cloud platform', async () => {
      // Arrange
      const FAKE_WEB_API = {
        config: () => FAKE_CONFIG,
      };
      const FAKE_QUERY_API = {
        getCapabilities: () => responses.salescloudCapabilities,
      };

      // Act
      // Assert
      const result = await expectSaga(sagas.loadConfigSaga, FAKE_ACTION, FAKE_GLOBAL)
        .provide([
          [matchers.call.fn(getWebApi), FAKE_WEB_API],
          [matchers.call.fn(FAKE_WEB_API.config), FAKE_CONFIG],
          [matchers.call.fn(getQueryApi), FAKE_QUERY_API],
          [matchers.call.fn(FAKE_QUERY_API.getCapabilities), responses.salescloudCapabilities],
        ])
        .run();

      expect(result.toJSON()).toMatchSnapshot();
    });
  });

  describe('loadFeatureTogglesSaga', () => {
    it('should fail if the Web API Client could not be retrieved', () => {
      // Arrange
      const FAKE_ACTION = actions.loadFeatureToggles();
      const FAKE_NULL_WEB_API = null;

      // Act
      // Assert
      return expectSaga(sagas.loadFeatureTogglesSaga, FAKE_ACTION)
        .provide([[matchers.call.fn(getWebApi), FAKE_NULL_WEB_API]])
        .not.put.like(actions.setFeatureToggles({}))
        .run();
    });

    it('should retrieve feature toggles from the API, and then dispatch a feature toggle set action', () => {
      // Arrange
      const FAKE_ACTION = actions.loadFeatureToggles();
      const FAKE_FEATURE_TOGGLES = {
        feature1: true,
      };
      const FAKE_WEB_API = {
        featureToggles: () => FAKE_FEATURE_TOGGLES,
      };

      // Act
      // Assert
      return expectSaga(sagas.loadFeatureTogglesSaga, FAKE_ACTION)
        .provide([[matchers.call.fn(getWebApi), FAKE_WEB_API], [matchers.call.fn(FAKE_WEB_API.featureToggles), FAKE_FEATURE_TOGGLES]])
        .call(getWebApi, FAKE_ACTION)
        .call.fn(FAKE_WEB_API.featureToggles)
        .put.like({ action: actions.setFeatureToggles(FAKE_FEATURE_TOGGLES) })
        .run();
    });

    it('should match the snapshot', async () => {
      // Arrange
      const FAKE_ACTION = correlate(actions.loadFeatureToggles(), 'CORRELATION_ID');
      const FAKE_FEATURE_TOGGLES = {
        feature1: true,
      };
      const FAKE_WEB_API = {
        featureToggles: () => FAKE_FEATURE_TOGGLES,
      };

      // Act
      const result = await expectSaga(sagas.loadFeatureTogglesSaga, FAKE_ACTION)
        .provide([[matchers.call.fn(getWebApi), FAKE_WEB_API], [matchers.call.fn(FAKE_WEB_API.featureToggles), FAKE_FEATURE_TOGGLES]])
        .run();

      // Assert
      expect(result.toJSON()).toMatchSnapshot();
    });
  });
});
