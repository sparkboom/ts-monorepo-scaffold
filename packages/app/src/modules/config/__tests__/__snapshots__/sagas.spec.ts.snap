// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Config > Sagas loadConfigSaga should match snapshot for Marketing Cloud platform 1`] = `
Object {
  "call": Array [
    Object {
      "@@redux-saga/IO": true,
      "combinator": false,
      "payload": Object {
        "args": Array [
          Object {
            "correlationId": "CORRELATION_ID",
            "type": "CONFIG__LOAD",
          },
        ],
        "context": null,
        "fn": "@@redux-saga-test-plan/json/function/getWebApi",
      },
      "type": "CALL",
    },
    Object {
      "@@redux-saga/IO": true,
      "combinator": false,
      "payload": Object {
        "args": Array [],
        "context": Object {
          "config": "@@redux-saga-test-plan/json/function/config",
        },
        "fn": "@@redux-saga-test-plan/json/function/config",
      },
      "type": "CALL",
    },
    Object {
      "@@redux-saga/IO": true,
      "combinator": false,
      "payload": Object {
        "args": Array [
          Object {
            "correlationId": "CORRELATION_ID",
            "type": "CONFIG__LOAD",
          },
        ],
        "context": null,
        "fn": "@@redux-saga-test-plan/json/function/getQueryApi",
      },
      "type": "CALL",
    },
    Object {
      "@@redux-saga/IO": true,
      "combinator": false,
      "payload": Object {
        "args": Array [],
        "context": Object {
          "getCapabilities": "@@redux-saga-test-plan/json/function/getCapabilities",
        },
        "fn": "@@redux-saga-test-plan/json/function/getCapabilities",
      },
      "type": "CALL",
    },
  ],
  "put": Array [
    Object {
      "@@redux-saga/IO": true,
      "combinator": false,
      "payload": Object {
        "action": Object {
          "correlationId": "CORRELATION_ID",
          "payload": Object {
            "details": Object {
              "status": "IN_PROGRESS",
            },
            "message": null,
            "relatedAction": Object {
              "correlationId": "CORRELATION_ID",
              "type": "CONFIG__LOAD",
            },
          },
          "type": "CONFIG__LOAD_STATUS",
        },
        "channel": undefined,
      },
      "type": "PUT",
    },
    Object {
      "@@redux-saga/IO": true,
      "combinator": false,
      "payload": Object {
        "action": Object {
          "correlationId": "CORRELATION_ID",
          "payload": Object {
            "option1": true,
          },
          "type": "CONFIG__SET",
        },
        "channel": undefined,
      },
      "type": "PUT",
    },
    Object {
      "@@redux-saga/IO": true,
      "combinator": false,
      "payload": Object {
        "action": Object {
          "correlationId": "CORRELATION_ID",
          "payload": Object {
            "platform": "MARKETING_CLOUD",
          },
          "type": "CONFIG__SET",
        },
        "channel": undefined,
      },
      "type": "PUT",
    },
    Object {
      "@@redux-saga/IO": true,
      "combinator": false,
      "payload": Object {
        "action": Object {
          "correlationId": "CORRELATION_ID",
          "payload": Object {
            "details": null,
            "message": null,
            "relatedAction": Object {
              "correlationId": "CORRELATION_ID",
              "type": "CONFIG__LOAD",
            },
          },
          "type": "CONFIG__LOAD_SUCCESS",
        },
        "channel": undefined,
      },
      "type": "PUT",
    },
  ],
}
`;

exports[`Config > Sagas loadConfigSaga should match snapshot for Sales Cloud platform 1`] = `
Object {
  "call": Array [
    Object {
      "@@redux-saga/IO": true,
      "combinator": false,
      "payload": Object {
        "args": Array [
          Object {
            "correlationId": "CORRELATION_ID",
            "type": "CONFIG__LOAD",
          },
        ],
        "context": null,
        "fn": "@@redux-saga-test-plan/json/function/getWebApi",
      },
      "type": "CALL",
    },
    Object {
      "@@redux-saga/IO": true,
      "combinator": false,
      "payload": Object {
        "args": Array [],
        "context": Object {
          "config": "@@redux-saga-test-plan/json/function/config",
        },
        "fn": "@@redux-saga-test-plan/json/function/config",
      },
      "type": "CALL",
    },
    Object {
      "@@redux-saga/IO": true,
      "combinator": false,
      "payload": Object {
        "args": Array [
          Object {
            "correlationId": "CORRELATION_ID",
            "type": "CONFIG__LOAD",
          },
        ],
        "context": null,
        "fn": "@@redux-saga-test-plan/json/function/getQueryApi",
      },
      "type": "CALL",
    },
    Object {
      "@@redux-saga/IO": true,
      "combinator": false,
      "payload": Object {
        "args": Array [],
        "context": Object {
          "getCapabilities": "@@redux-saga-test-plan/json/function/getCapabilities",
        },
        "fn": "@@redux-saga-test-plan/json/function/getCapabilities",
      },
      "type": "CALL",
    },
  ],
  "put": Array [
    Object {
      "@@redux-saga/IO": true,
      "combinator": false,
      "payload": Object {
        "action": Object {
          "correlationId": "CORRELATION_ID",
          "payload": Object {
            "details": Object {
              "status": "IN_PROGRESS",
            },
            "message": null,
            "relatedAction": Object {
              "correlationId": "CORRELATION_ID",
              "type": "CONFIG__LOAD",
            },
          },
          "type": "CONFIG__LOAD_STATUS",
        },
        "channel": undefined,
      },
      "type": "PUT",
    },
    Object {
      "@@redux-saga/IO": true,
      "combinator": false,
      "payload": Object {
        "action": Object {
          "correlationId": "CORRELATION_ID",
          "payload": Object {
            "option1": true,
          },
          "type": "CONFIG__SET",
        },
        "channel": undefined,
      },
      "type": "PUT",
    },
    Object {
      "@@redux-saga/IO": true,
      "combinator": false,
      "payload": Object {
        "action": Object {
          "correlationId": "CORRELATION_ID",
          "payload": Object {
            "platform": "SALES_CLOUD",
          },
          "type": "CONFIG__SET",
        },
        "channel": undefined,
      },
      "type": "PUT",
    },
    Object {
      "@@redux-saga/IO": true,
      "combinator": false,
      "payload": Object {
        "action": Object {
          "correlationId": "CORRELATION_ID",
          "payload": Object {
            "details": null,
            "message": null,
            "relatedAction": Object {
              "correlationId": "CORRELATION_ID",
              "type": "CONFIG__LOAD",
            },
          },
          "type": "CONFIG__LOAD_SUCCESS",
        },
        "channel": undefined,
      },
      "type": "PUT",
    },
  ],
}
`;

exports[`Config > Sagas loadFeatureTogglesSaga should match the snapshot 1`] = `
Object {
  "call": Array [
    Object {
      "@@redux-saga/IO": true,
      "combinator": false,
      "payload": Object {
        "args": Array [
          Object {
            "correlationId": "CORRELATION_ID",
            "type": "CONFIG__FEATURE_TOGGLES_LOAD",
          },
        ],
        "context": null,
        "fn": "@@redux-saga-test-plan/json/function/getWebApi",
      },
      "type": "CALL",
    },
    Object {
      "@@redux-saga/IO": true,
      "combinator": false,
      "payload": Object {
        "args": Array [],
        "context": Object {
          "featureToggles": "@@redux-saga-test-plan/json/function/featureToggles",
        },
        "fn": "@@redux-saga-test-plan/json/function/featureToggles",
      },
      "type": "CALL",
    },
  ],
  "put": Array [
    Object {
      "@@redux-saga/IO": true,
      "combinator": false,
      "payload": Object {
        "action": Object {
          "correlationId": "CORRELATION_ID",
          "payload": Object {
            "featureToggles": Object {
              "feature1": true,
            },
          },
          "type": "CONFIG__FEATURE_TOGGLES_SET",
        },
        "channel": undefined,
      },
      "type": "PUT",
    },
  ],
}
`;
